src/pip/_internal/index.py

```diff
@@ -594,21 +604,20 @@ class FoundCandidates(object):
 
     def __init__(
         self,
-        candidates,     # type: List[InstallationCandidate]
-        versions,       # type: Set[str]
-        evaluator,      # type: CandidateEvaluator
+        candidates,             # type: List[InstallationCandidate]
+        applicable_candidates,  # type: List[InstallationCandidate]
+        evaluator,              # type: CandidateEvaluator
     ):
         # type: (...) -> None
         """
         :param candidates: A sequence of all available candidates found.
-        :param versions: The applicable versions to filter applicable
-            candidates.
+        :param applicable_candidates: The applicable candidates.
         :param evaluator: A CandidateEvaluator object to sort applicable
             candidates by order of preference.
         """
+        self._applicable_candidates = applicable_candidates
         self._candidates = candidates
         self._evaluator = evaluator
-        self._versions = versions
```

Comment:
Late nit: I'd put an `assert set(self._applicable_candidates) <= set(self._candidates)` to make sure args are coherent.

Reply:
Thanks for taking a look, @xavfernandez. I'll do this in a future PR.